app/Models/Multiplayer/Event.php

```diff
@@ -37,4 +48,23 @@ public function game()
     {
         return $this->belongsTo(Game::class);
     }
+
+    public function user()
+    {
+        return $this->belongsTo(User::class);
+    }
+
+    public function scopeDefault($query)
+    {
+        return $query->orderBy('event_id', 'asc');
+    }
+
+    public function getTextAttribute($value)
+    {
+        if (array_key_exists($value, self::EVENT_TYPES)) {
```

Comment:
```php
if (...) {
  return ['type' => static::EVENT_TYPES[...]];
} else {
  return ['type' => 'other', 'text' => $value];
}
```

(though probably should be under different attribute/function name (something like `detail`?)